packages/teleterm/src/ui/DocumentTerminal/useDocumentTerminal.ts

```diff
@@ -58,16 +58,16 @@ async function initState(
   docsService: DocumentsService,
   doc: Doc
 ) {
-  const getClusterActualName = () => {
+  const getClusterName = () => {
     const cluster = ctx.clustersService.findCluster(clusterUri);
     if (cluster) {
-      return cluster.actualName;
+      return cluster.name;
     }
 
     /*
      When restoring the documents, we do not always have the leaf clusters already fetched.
      In that case we can fall back to `clusterId` from a leaf cluster URI
-     (for a leaf cluster `clusterId` === `actualName`)
+     (for a leaf cluster `clusterId` === `name`)
     */
     const parsed = routing.parseClusterUri(clusterUri);
```

Comment:
That damn hack lol. But maybe it gives us some insight for the future – we should strive to make it so that `Document` holds all information that's required to connect to a resource without having to wait for data from another source.